tests/client_decode.rs

```diff
@@ -0,0 +1,68 @@
+mod client_decode {
+    use async_h1::client;
+    use async_std::io::Cursor;
+    use http_types::headers;
+    use http_types::Result;
+    use pretty_assertions::assert_eq;
+
+    async fn decode_str(s: &'static str) -> http_types::Result<http_types::Response> {
+        client::decode(Cursor::new(s.replace("\n", "\r\n"))).await
```

Comment:
It may be easier to read these raw strings if they are e.g. 
```
vec![
    "HTTP/1.1 200 OK",
    "transfer-encoding: chunked",
    "content-type: text/plain",
    "",
    "",
];
```

with some kind of `map(|s| s.append("\r\n")` and then a `join("")`.

Reply:
what do you think about https://github.com/dtolnay/indoc

Reply:
Hmm, better, although the trailing linebreaks (and any special white space) may become less clear?

Reply:
wasn't sure about this, but i think it ended up an improvement, which strikes me as a language issue: 59caa84